src/hearings/containers/cancel-hearing/cancel-hearing.component.ts

```diff
@@ -86,10 +83,17 @@ export class CancelHearingComponent implements OnInit {
   }
 
   public executeContinue(): void {
+    const cancellationErrorMessage = 'There was a system error and your request could not be processed. Please try again.';
     if (this.isFormValid()) {
-      this.hearingsService.cancelHearingRequest(this.hearingId, this.getChosenReasons()).subscribe(() => {
-        return this.router.navigate(['cases', 'case-details', this.caseId, 'hearings']).then();
-      });
+      this.hearingsService.cancelHearingRequest(this.hearingId, this.getChosenReasons()).subscribe(
```

Comment:
could we add something in the mock cancelHearingRequest API, i.e. if the user choose a special reason(like reason3 code) then it throws the error and this code catch it and Vamshi can test with this scenario.

Reply:
@SivaprakasamSivaprakasam even in integration testing we can't guarantee the backend can return non 200 error, so ideally we should have a workaround in the mock API to show different error code, will speak to you tomorrow to how to achieve this.